wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/pyramidannotation/PyramidAnnotationDataHandler.java

```diff
@@ -0,0 +1,112 @@
+import gov.nist.itl.ssd.wipp.backend.data.pyramidannotation.timeslices.PyramidAnnotationTimeSlice;
+
+/**
+ * @author Mohamed Ouladi <mohamed.ouladi at nist.gov>
+ */
+@Component("pyramidAnnotationDataHandler")
+public class PyramidAnnotationDataHandler extends BaseDataHandler implements DataHandler {
+
+    @Autowired
+    CoreConfig config;
+    
+    @Autowired
+    private PyramidAnnotationRepository pyramidAnnotationRepository;
+
+    public PyramidAnnotationDataHandler() {
+    }
+		
+    @Override
+    public void importData(Job job, String outputName) throws IOException, JobExecutionException {
+
+        String pyramidAnnotationFilenamePattern =
+        		PyramidAnnotationConfig.PYRAMID_ANNOTATION_FILENAME_PREFIX
+                        + "([0-9]+)"
+                        + PyramidAnnotationConfig.PYRAMID_ANNOTATION_FILENAME_SUFFIX;
+        File jobTempOutputFolder = getJobOutputTempFolder(job.getId(), outputName);
+
+        List<PyramidAnnotationTimeSlice> timeSlices = Stream
+                .of(jobTempOutputFolder.listFiles((d, name) -> name.matches(pyramidAnnotationFilenamePattern)))
+                .map(f -> createTimeSlice(f.getName()))
+                .collect(Collectors.toList());
+        
+        PyramidAnnotation pyramidAnnotation = new PyramidAnnotation(job, timeSlices, outputName);
+        // We save so that an Id is generated.
+        pyramidAnnotationRepository.save(pyramidAnnotation);
+
+        new File(config.getPyramidAnnotationsFolder()).mkdirs();
+        File pyramidAnnotationFolder = new File(
+                config.getPyramidAnnotationsFolder(), pyramidAnnotation.getId());
+        boolean success = getJobOutputTempFolder(job.getId(), outputName).renameTo(pyramidAnnotationFolder);
+        if (!success) {
+        	pyramidAnnotationRepository.delete(pyramidAnnotation);
+            throw new JobExecutionException(
+                    "Cannot move pyramid annotation to final destination.");
+        }
+        
+        setOutputId(job, outputName, pyramidAnnotation.getId());
+    }
+	
+
+	@Override
+	public String exportDataAsParam(String value) {
+        String pyramidAnnotationId = value;
+        String pyramidAnnotationPath;
+
+        // check if the input of the job is the output of another job and if so return the associated path
+        String regex = "\\{\\{ (.*)\\.(.*) \\}\\}";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher m = pattern.matcher(pyramidAnnotationId);
+        if (m.find()) {
+            String jobId = m.group(1);
+            String outputName = m.group(2);
+            pyramidAnnotationPath = getJobOutputTempFolder(jobId, outputName).getAbsolutePath();
+        }
+        // else return the path of the notebook
+        else {
+            File notebookFolder = new File(config.getPyramidAnnotationsFolder(), pyramidAnnotationId);
+            pyramidAnnotationPath = notebookFolder.getAbsolutePath();
```

Comment:
Should be pyramidAnnotation instead of notebook

Reply:
fixed